app/components/swap/swap.tsx

```diff
@@ -0,0 +1,293 @@
+        >
+          <div className="relative h-[15.95px] w-[15.95px] md:h-[32px] md:w-[32px]">
+            <Image
+              src={
+                theme === "dark" ? "/icons/swap.svg" : "/icons/swap-white.svg"
+              }
+              alt="Swap"
+              fill
+              style={{ objectFit: "contain" }}
+            />
+          </div>
+        </button>
+
+        <div className="mt-0 flex w-full flex-col">
+          <label
+            className={`mb-2 text-[9.97px] md:text-[16px] ${theme === "dark" ? "text-black" : "text-white"}`}
+          >
+            To
+          </label>
+          <div
+            className={`flex h-[78.49px] flex-col items-center rounded-[8px] px-[10px] py-[10px] md:h-[134px] md:px-[24px] md:py-[30px] ${theme === "dark" ? "border border-[#dadada] bg-[#f5f5f5]" : "border border-grey-700 bg-grey-600"}`}
+          >
+            <div className="flex w-full items-center justify-between">
+              <div className="flex flex-col items-start">
+                <input
+                  type="text"
+                  value={equivalent}
+                  readOnly
+                  className={`w-[45%] bg-transparent text-[18.59px] font-[700] outline-none md:w-[75%] md:text-[32px] ${theme === "dark" ? "text-black" : "text-white"}`}
+                />
+                <p className="ml-[2px] max-w-[45%] overflow-hidden text-ellipsis whitespace-nowrap text-[9.97px] font-[600] text-grey-500 md:text-[16px]">
+                  = ${equivalent}
+                </p>
+              </div>
+              <CustomSelect
+                selectedToken={toToken}
+                onTokenSelect={setToToken}
+                theme={theme}
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <div className="flex w-full flex-col gap-2 md:my-2">
+        <div className="flex w-full items-center justify-between">
+          <p
+            className={`${theme === "dark" ? "" : ""} text-[9.97px] font-[700] text-grey-500 md:text-[16px]`}
+          >
+            Current price
+          </p>
+          <p
+            className={`${theme === "dark" ? "text-black" : "text-white"} text-[9.97px] font-[700] md:text-[16px]`}
+          >
+            1 {fromToken} = {rate.toFixed(6)} {toToken}
+          </p>
+        </div>
+        <div className="flex w-full items-center justify-between">
+          <p
+            className={`${theme === "dark" ? "" : ""} text-[9.97px] font-[700] text-grey-500 md:text-[16px]`}
+          >
+            Min received
+          </p>
+          <p
+            className={`${theme === "dark" ? "text-black" : "text-white"} text-[9.97px] font-[700] md:text-[16px]`}
+          >
+            {equivalent} {toToken}
+          </p>
+        </div>
+      </div>
+
+      {error && (
+        <p className="text-[9.97px] text-red-500 md:text-[16px]">{error}</p>
+      )}
+
+      <button
+        onClick={address ? handleSwap : handleConnectWallet}
```

Comment:
No need for the connect wallet function, Swap button should be disabled if not connected